django/src/rdwatch/views/site_observation.py

```diff
@@ -187,14 +246,72 @@ def cancel_site_observation_images(request: HttpRequest, pk: int):
                 fetching_task.celery_id = ''
                 fetching_task.save()
             else:
-                return Response(
-                    f'There is no running task for Site Observation Id: {pk}',
-                    status=status.HTTP_409_CONFLICT,
+                return (
+                    409,
+                    f'There is no running task for Observation Id: {evaluation_id}',
                 )
 
         else:
-            return Response(
-                f'There is no running task for Site Observation Id: {pk}', status=404
+            return (
+                404,
+                f'There is no running task for Observation Id: {evaluation_id}',
             )
 
-    return Response(status=202)
+    return 202, True
+
+
+@router.patch('/{id}/')
+def update_site_observation(
+    request: HttpRequest, id: int, data: SiteObservationRequest
+):
+    site_observation = get_object_or_404(SiteObservation, pk=id)
+    # create a copy of it in the history log
+    SiteObservationTracking.objects.create(
+        timestamp=site_observation.timestamp,
+        geom=site_observation.geom,
+        score=site_observation.score,
+        siteval=site_observation.siteeval,
+        notes=site_observation.notes,
+        edited=datetime.now(),
+        observation=site_observation.pk,
+    )
+
+    if data.label:
+        site_observation.label = lookups.ObservationLabel.objects.get(slug=data.label)
+    if data.notes:
+        site_observation.notes = data.notes
+    site_observation.timestamp = data.timestamp
+    if data.spectrum:
+        site_observation.spectrum = data.spectrum
+    if data.constellation:
+        site_observation.constellation = lookups.Constellation.objects.get(
+            slug=data.label
+        )
+
+    site_observation.save()
+
+    return site_observation
```

Comment:
Similar situation here as with site eval above -
```suggestion
    with transaction.atomic():
        site_observation = get_object_or_404(SiteObservation.objects.select_for_update(), pk=id)
        # create a copy of it in the history log
        SiteObservationTracking.objects.create(
            timestamp=site_observation.timestamp,
            geom=site_observation.geom,
            score=site_observation.score,
            siteval=site_observation.siteeval,
            notes=site_observation.notes,
            edited=datetime.now(),
            observation=site_observation.pk,
        )

        if data.label:
            site_observation.label = lookups.ObservationLabel.objects.get(slug=data.label)
        if data.notes:
            site_observation.notes = data.notes
        site_observation.timestamp = data.timestamp
        if data.spectrum:
            site_observation.spectrum = data.spectrum
        if data.constellation:
            site_observation.constellation = lookups.Constellation.objects.get(
                slug=data.label
            )

        site_observation.save()

        return site_observation
```